csharp/src/Apache.Arrow/Ipc/MessageSerializer.cs

```diff
@@ -73,13 +73,27 @@ internal static Schema GetSchema(Flatbuf.Schema schema)
             return new Schema(fields, metadata, copyCollections: false);
         }
 
-        private static Field FieldFromFlatbuffer(Flatbuf.Field flatbufField)
+        private static Field FieldFromFlatbuffer(Flatbuf.Field flatbufField, LazyCreator<DictionaryMemo> lazyDictionaryMemo)
         {
             Field[] childFields = flatbufField.ChildrenLength > 0 ? new Field[flatbufField.ChildrenLength] : null;
             for (int i = 0; i < flatbufField.ChildrenLength; i++)
             {
                 Flatbuf.Field? childFlatbufField = flatbufField.Children(i);
-                childFields[i] = FieldFromFlatbuffer(childFlatbufField.Value);
+                childFields[i] = FieldFromFlatbuffer(childFlatbufField.Value, lazyDictionaryMemo);
+            }
+
+            Flatbuf.DictionaryEncoding? de = flatbufField.Dictionary;
+            IArrowType type = GetFieldArrowType(flatbufField, childFields);
+
+            if (de.HasValue)
+            {
+                Flatbuf.Int? indexTypeAsInt = de.Value.IndexType;
+                if (!indexTypeAsInt.HasValue)
+                {
+                    throw new InvalidDataException("Dictionary type not defined");
```

Comment:
Would this be a more correct message?

```suggestion
                    throw new InvalidDataException("Dictionary IndexType not defined");
```

Reply:
I fixed it.